packages/docs/modules/ROOT/pages/network-js.adoc

```diff
@@ -0,0 +1,302 @@
+We then get a web3Context using `useWeb3` which will attempt to retrieve an injected web3 provider (e.g. MetaMask), otherwise it will fall back to a network connection, which in this case is a connection to mainnet via Infura.
+The `networkId`, `networkName` and `providerName` are obtained from the web3Context and displayed in the dapp.
+
+On the `client/src/App.js` file, replace the placeholder code in `App.js` in our react project with the following code:
+[source,js]
+----
+import React from 'react';
+import './App.css';
+
+import { useWeb3 } from '@openzeppelin/network/react';
+
+const infuraProjectId = '95202223388e49f48b423ea50a70e336';
+
+function App() {
+
+  const web3Context = useWeb3(`wss://mainnet.infura.io/ws/v3/${infuraProjectId}`);
+  const { networkId, networkName, providerName } = web3Context;
+
+  return (
+    <div className="App">
+      <div>
+        <h1>OpenZeppelin Network.js</h1>
+        <div>Network: {networkId ? `${networkId} – ${networkName}` : 'No connection'}</div>
+        <div>Provider: {providerName}</div>
+      </div>
+    </div>
+  );
+}
+
+export default App;
+----
+
+We can start our dapp by running `npm start` from within the `client` directory. 
+
+The dapp will display the Ethereum network ID and network name, either of mainnet from our network connection if no injected web3 provider such as MetaMask is available, otherwise it will display the current network as selected by the injected web3 provider, along with the web3 provider name.
+
+We use `useWeb3` which attempts to obtain an injected web3 provider first, before falling back to a network connection.  Alternatively we could use `useWeb3Injected` for an injected web3 provider or `useWeb3Network` for a network provider e.g. Infura or a private node.  Dapps can also be configured to use multiple providers if necessary.
+
+NOTE: If you use Infura in your own dapp, you will need to create an account with Infura and setup an Infura Project (with corresponding Infura Project ID).  See https://infura.io[Infura website] to create an account.
+
+== Add a component to the dapp
+For the second iteration of our dapp we will move the display of the current Ethereum network to a component and show how components are re-rendered when changes are made, such as the network.
+
+To install MetaMask as our injected web3 provider (if you don't have an injected web3 provider installed already), follow the instructions on the https://metamask.io[MetaMask website] to install the browser extension.
+
+In the code below, our `Web3Info` component expects a web3Context.  The `networkId`, `networkName` and `providerName` are obtained from the web3Context and displayed in the component.
+
+In the `client/src` directory create a `components` directory. In the `components` directory create a `Web3Info` directory for our `Web3Info` component.  Then create an `index.js` file in the `Web3Info` directory with the following code:
```

Comment:
This is equivalent and much simpler.

```suggestion
In the `client/src` directory create a `components` directory. In the `components` directory create a `Web3Info.js` file  with the following code:
```

Reply:
Shows how little I currently know about React components :smile: